code/menuui/techmenu.cpp

```diff
@@ -896,35 +895,43 @@ void techroom_change_tab(int num)
 		case INTEL_DATA_TAB:
 
 			// load intel if necessary
-			if ( Intel_loaded == 0 ) {
-				// now populate the entry structs
+			if ( !Intel_loaded ) {
+				if (Intel_list == nullptr) {
+					Intel_list = new tech_list_entry[Intel_info.size()];
```

Comment:
Any particular reason to use dynamic memory here instead of a vector? I looked at the relevant code and nothing jumped out at me to suggest that we need that dynamic memory here.

Reply:
Two reasons:
1) The ship and weapons lists are handled the same way
2) `Current_list` points to whatever list is being handled in the tech room.  That is of type `tech_list_entry *` so the intel list needs to match it

Reply:
You can still get a pointer to the contents of a vector but this works as well.